src/UnitOfWork/OperationConsolidationMode.php

```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types = 1);
+
+namespace BrandEmbassy\UnitOfWork;
+
+/**
+ * @final
+ */
+class OperationConsolidationMode
+{
+    public function __construct(
+        private readonly bool $isLoggingEnabled,
+        private readonly bool $isDryRunUnlimitedConsolidation,
+        private readonly bool $isUnlimitedConsolidation
```

Comment:
what about to have defaults here? later you may use named arguments to easily set perhaps only the isUnlimitedConsolidation to true and the others will be off
```suggestion
        private readonly bool $isLoggingEnabled = false,
        private readonly bool $isDryRunUnlimitedConsolidation = false,
        private readonly bool $isUnlimitedConsolidation = false,
```